src/classes/STG_PanelOppNaming_CTRL.cls

```diff
@@ -183,21 +197,50 @@ public with sharing class STG_PanelOppNaming_CTRL extends STG_Panel {
     }
 
     /*********************************************************************************************************
-    * @description The list of SelectOptions for Opportunity Attribution 
+    * @description The list of SelectOptions for Opportunity Attribution
     */
-    static public list<SelectOption> listSOOpportunityAttribution {
-        get {    
+    public static list<SelectOption> listSOOpportunityAttribution {
+        get {
             if (listSOOpportunityAttribution == null) {
-                listSOOpportunityAttribution = new list<SelectOption>();
-                listSOOpportunityAttribution.add(new SelectOption(Label.oppNamingBoth, Label.oppNamingBoth));
-                listSOOpportunityAttribution.add(new SelectOption(Label.oppNamingIndividual, Label.oppNamingIndividual));
-                listSOOpportunityAttribution.add(new SelectOption(Label.oppNamingOrganizational, Label.oppNamingOrganizational));
+                listSOOpportunityAttribution = getOppAttributionOptions();
```

Comment:
All attribution label determination and retrieval can be handled by a new inner class `AttributionSettings`. If so, then this line can be 
`listSOOpportunityAttribution = new AttributionSettings().getSelectOptions();`
and new methods below can be removed.

Similarly the `OPP_OpportunityNaming`:
- line 191: `STG_PanelOppNaming_CTRL.AttributionSettings attributionSettings = new STG_PanelOppNaming_CTRL.AttributionSettings()
                .withSelected(ons.Attribution__c);`
- instead of `if (oppAttributionValue == Label.oppNamingBoth)` can use `if (attributionSettings.isBoth())`
- `oppAttributionValue` on lines 199 and 208 can use `attributionSettings.selectedLabel`.


```
/***
    * @description Handles Opportunity Naming Settings Attribution options
    */
    public class AttributionSettings {

        public String selectedOption { get; private set; }
        public String selectedLabel { get; private set; }

        /***
        * @description Sets attribution value as configured in the Opportunity Naming Settings
        */
        public AttributionSettings withSelected(String selectedOption) {
            this.selectedOption = selectedOption;
            this.selectedLabel = getLabel(this.selectedOption);
            return this;
        }

        /***
        * @description Contains Attribution API name to custom label mapping
        */
        private Map<String, String> labelByOption {
            get {
                if (labelByOption == null) {
                    labelByOption = new Map<String, String>{
                        AttributionOptions.Both.name() => System.Label.OppNamingBoth,
                        System.Label.OppNamingBoth => System.Label.OppNamingBoth,
                        'Both' => System.Label.OppNamingBoth,

                        AttributionOptions.Contact_Donations.name() => System.Label.oppNamingIndividual,
                        System.Label.oppNamingIndividual => System.Label.oppNamingIndividual,
                        'Contact Donations' => System.Label.oppNamingIndividual,

                        AttributionOptions.Organization_Donations.name() => System.Label.oppNamingOrganizational,
                        System.Label.oppNamingOrganizational => System.Label.oppNamingOrganizational,
                        'Organization Donations' => System.Label.oppNamingOrganizational
                    };
                }
                return labelByOption;
            }
            set;
        }

        /***
        * @description Generate the select options for attribution settings
        */
        public List<SelectOption> getSelectOptions() {
            List<SelectOption> options = new List<SelectOption>();

            for (AttributionOptions option : AttributionOptions.values()) {
                options.add(new SelectOption(option.name(), getLabel(option.name())));
            }

            return options;
        }

        /***
        * @description Map the attribution option to the custom label used in Opp naming or settings field option display
        */
        public String getLabel(String option) {
            String label = labelByOption.get(option);

            return String.isBlank(label) ? '' : label;
        }

        /***
        * @description Returns true if attribution is applicable to both Contact and Organization Donations
        */
        public Boolean isBoth() {
            return selectedLabel == System.Label.OppNamingBoth;
        }

    }
```